raster-foundry/raster-foundry
# Add bands jsonb property to datasources

## Overview
This attribute is pretty much entirely for use on the frontend, as the backend
only cares about the actual index values. Any error handling and validations
should therefore also happen on the frontend.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

## TODO
* fix tests

## Testing Instructions
* Create a new datasource
 * Verify that you can make a put request to the datasource and update its bands.
Example request body: 
```json
{
      "id": "4a50cb75-815d-4fe5-8bc1-144729ce5b42",
      "createdAt": "2017-06-01T18:55:18.011416Z",
      "createdBy": "default",
      "modifiedAt": "2017-06-01T18:55:18.011416Z",
      "modifiedBy": "default",
      "owner": "default",
      "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
      "name": "Sentinel-2",
      "visibility": "PUBLIC",
      "composites": {
        "invalid": {
          "label": "Invalid Bands",
          "value": {
            "redBand": 0,
            "blueBand": 0,
            "greenBand": 0
          }
        }
      },
      "extras": {},
      "bands": {"1": 1, "2": 2, "3a": 3, "3b": 4}
}
```
* Verify that you cannot modify datasources that you do not own unless you are in the root org

Closes #2324
